client/src/components/pages/ViewActivities/index.js

```diff
@@ -1,18 +1,50 @@
+import axios from 'axios';
 import React, { Component } from "react";
-import "./index.css";
 import {Link} from 'react-router-dom'
 import Header from "../../Header";
-
+import Table from "../../Table";
+import {StyledPage, StyledTilte} from "./index.style";
 
 
 class ViewActivities extends Component {
+
+  state={
+    result:[],
+  }
+
+  getColumnsList(){
```

Comment:
This doesn't need to be a function on the class, instead it can go below, inside the render function, and you can just pass that constant as the prop (instead of calling this function):

```js
render() {
  const columns = [{ Header:'id',accessor:'__id',show:false}, ... ]
  return (
      <React.Fragment>
        <Header/>
        <StyledPage>
          <StyledTilte> جميع الانشطة </StyledTilte>
          <Table columns={columns} data={this.state.result}/>
          <li><Link to="/CreateActivity">اضافة نشاط جديد</Link></li>
        </StyledPage>
      </React.Fragment>
```

Reply:
done